### PR TITLE
Allow running command on remote servers / SSH

### PR DESCRIPTION
**Pull request by pyrech**
Just a POC for now

### CODE REVIEW

File: doc/01-installation.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 ## Installation
 
 > **Note**
-> Please note that castor needs PHP >= 8.1 to run.
+> Castor needs PHP >= 8.1 to run.
```
Comment (lyrixx) [inline doc/01-installation.md:6]: Your repo was outdated?

File: doc/11-ssh.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# SSH and remote servers
+
+Castor support running commands on remote servers through SSH. You can configure
```
Comment (lyrixx) [inline doc/11-ssh.md:3]:
```suggestion
Castor supports running commands on remote servers through SSH. You can configure
```

File: src/Context.php [hunk 3]
```diff
@@ -184,6 +223,58 @@ public function withVerbosityLevel(VerbosityLevel $verbosityLevel): self
...
+            $this->notify,
+            $this->verbosityLevel,
+            'localhost',
+            null,
+            [],
+        );
+    }
+
+    /**
+     * @param array{
```
Comment (lyrixx) [inline src/Context.php:252]: You can define a type to avoid repeating yourself. See the PHP Stan doc

File: src/functions.php [hunk 4]
```diff
@@ -97,6 +98,28 @@ function run(
         $context = $context->withNotify($notify);
     }
 
+    if (!\in_array($context->host, ['local', 'localhost', '127.0.0.1']) && $context->user) {
```
Comment (lyrixx) [inline src/functions.php:101]: You should add argument on the run command too

**Review by lyrixx — COMMENTED** [review]
I have some doubts about this feature. Do we really want to maintain it? Is it really bullets proof? (Maybe add a message to say it's experimental and we may break the API)

Thanks for the work anyway 👍🏼

**Review by lyrixx — COMMENTED** [review]
I have some doubts about this feature. Do we really want to maintain it? Is it really bullets proof? (Maybe add a message to say it's experimental and we may break the API)

Thanks for the work anyway 👍🏼

File: doc/01-installation.md (hunk 1 above)
Comment (pyrech) [reply doc/01-installation.md:6]: Nope, it was fixed on the readme but not on the doc

**Comment by pyrech** [comment]
There is some issue with tty so definitely not bullet proof on the interactivity part. I wonder if the use a real PHP implementation of a SSH client would be better to handle this or not.

I agree to mark this feature as experimental :+1:

**Comment by pyrech** [comment]
PR updated to only provide an `ssh()` function and mark it as experimental

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,7 @@
 * Display warning and update instructions when a new version is available
 * Better error reporting when a call to run() fail or when import() is not possible
 * Fix stubs generation
+* Add `ssh()` function to run commands on remote server via SSH
```
Comment (lyrixx) [inline CHANGELOG.md:9]:
```suggestion
* Add `ssh()` function to run commands on remote server via SSH
* Display warning and update instructions when a new version is available
* Better error reporting when a call to run() fail or when import() is not possible
* Fix stubs generation
```

File: doc/01-installation.md (hunk 1 above)
Comment (lyrixx) [inline doc/01-installation.md:6]:
```suggestion
> Castor requires PHP >= 8.1 to run.
```

File: doc/11-ssh.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
+# SSH and remote servers
+
+Castor supports running commands on remote servers through SSH with the `ssh()`
+function:
+
+```php
+use Castor\Attribute\AsTask;
+use function Castor\ssh;
```
Comment (lyrixx) [inline doc/11-ssh.md:8]:
```suggestion

use function Castor\ssh;
```

File: examples/ssh.php [hunk 7]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace ssh;
+
+use Castor\Attribute\AsTask;
+
+use function Castor\ssh;
+
+#[AsTask(description: 'Runs a command on a remote server')]
```
Comment (lyrixx) [inline examples/ssh.php:9]:
```suggestion
#[AsTask(description: 'Lists content of /var/www directory on the remote server')]
```

**Review by lyrixx — COMMENTED** [review]
LGTM, except few little comments

**Comment by lyrixx** [comment]
Can you rebase in order to fix conflicts?

**Review by lyrixx — APPROVED** [review]

**Pull request merged**